scripts/plot.py

```diff
@@ -0,0 +1,147 @@
+      tmp_num_sources, _, tmp_errors = parse_benchmark(input_file, algorithm, 'ArborX::BVH<'+backends[i]+'>', 'stddev', geometry)
+      num_sources.append(tmp_num_sources)
+      timings.append(tmp_timings)
+      errors.append(tmp_errors)
+      unified_num_sources = unified_num_sources + tmp_num_sources
+
+    unified_num_sources = list(set(unified_num_sources))
+    unified_num_sources.sort()
+    num_source_labels = [None] * len(unified_num_sources)
+    for i in range(len(unified_num_sources)):
+        num_source_labels[i] = '$10^' + str(round(math.log(unified_num_sources[i], 10))) + '$'
+
+    # Transform given timings into the plotting data
+    data = np.zeros([len(backends), len(unified_num_sources)])
+    errors_data = np.zeros([len(backends), 2, len(unified_num_sources)])
+    for i in range(0, len(backends)):
+        for j in range(0, len(num_sources[i])):
+            index = unified_num_sources.index(num_sources[i][j])
+            data[i, index] = (num_sources[i][j] / timings[i][j])/10**6
+            errors_data[i, 0, index] = num_sources[i][j]/10**6*(1/timings[i][j] - 1/(timings[i][j]+errors[i][j]))
+            errors_data[i, 1, index] = num_sources[i][j]/10**6*(1/(timings[i][j]-errors[i][j]) - 1/timings[i][j])
+
+    # Phase 2: plot generation
+    colors = tableau20()
+
+    ax = plt.subplot(111)
+
+    def autolabel(rects):
+        """Attach a text label above each bar in *rects*, displaying its height."""
+        for rect in rects:
+            height = round(rect.get_height(),1)
+            ax.annotate('{}'.format(height),
+                        xy=(rect.get_x() + rect.get_width() / 2, height-rect.get_height()/2),
+                        xytext=(0, 3),  # 3 points vertical offset
+                        textcoords="offset points",
+                        ha='center', va='bottom')
+
+    width = 0.6
+    scale = 1/len(backends)
+    for i in range(0, len(backends)):
+        barplot = ax.bar(np.arange(len(unified_num_sources)) + scale*(i-1.5) * width,
+                         width=scale*width*np.ones(len(unified_num_sources)),
+                         height=data[i, :],
+                         yerr=errors_data[i, :],
+                         capsize=5,
+                         edgecolor='k',
+                         label=backends[i])
+        if numbers:
+            autolabel(barplot)
+
+    ax.legend(loc='upper left', fontsize=20)
+
+    ax.set_xlabel('Number of objects indexed', fontsize=20)
+    plt.xticks(fontsize=18)
+    ax.set_xticks(np.arange(len(num_source_labels)))
+    ax.set_xticklabels(num_source_labels)
+    ax.tick_params(axis='x', which='major', bottom='off', top='off')
+    ax.set_xlim([-0.5, len(unified_num_sources)-0.5])
+
+    ax.set_ylabel('Rate (million objects/second)', fontsize=20)
+    plt.yticks(fontsize=18)
+    ax.yaxis.grid('on')
+    plt.title('Construction', fontsize=18)
```

Comment:
This should not be hardcoded, and instead should be mapped from the `algorithm`.